src/librustc_mir/util/borrowck_errors.rs

```diff
@@ -284,22 +298,25 @@ pub trait BorrowckErrors {
         self.cancel_if_wrong_origin(err, o)
     }
 
-    fn cannot_assign(&self, span: Span, desc: &str, o: Origin) -> DiagnosticBuilder
+    fn cannot_assign(self, span: Span, desc: &str, o: Origin) -> DiagnosticBuilder<'cx>
+        where Self: Sized + Copy
```

Comment:
...then you could remove them from all these places

Reply:
@nikomatsakis 
the compiler complains that the `self` in trait should be `Sized + Copy`, and I found that `TyCtxt` already `#[derive(Copy, Clone)]`, so I rewrote `pub trait BorrowckError<'tcx>: Sized + Copy {}` instead of appending `where` clause in every method. Is this make sense?

Reply:
Yep! That's what I was proposing, fact.